pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -313,14 +330,17 @@ func (r *OAMApplicationReconciler) Reconcile(req reconcile.Request) (result reco
 	r.updateStatus(ctx, ac, acPatch, workloads)
 
 	ac.Status.Dependency = v1alpha2.DependencyStatus{}
-	waitTime := longWait
 	if len(depStatus.Unsatisfied) != 0 {
-		waitTime = dependCheckWait
 		ac.Status.Dependency = *depStatus
+		return reconcile.Result{RequeueAfter: dependCheckWait}, nil
 	}
 
 	// the posthook function will do the final status update
-	return reconcile.Result{RequeueAfter: waitTime}, nil
+	if r.limitedReconcile {
+		return reconcile.Result{}, nil
+	}
+	// enforce to reconcile periodically, even no event occurs
+	return reconcile.Result{RequeueAfter: longWait}, nil
```

Comment:
We should not implement like this.

We should check if the spec is unchanged and already reconciled succeed.

I assume at least we should have these conditions:

1. metadata.generation==status.observedGeneration.
2. the component spec(revision) is the same with the living workload revision.
3. the status is reconciled succeed
4. the workload is health
5. the trait CR is working normally (I assume we can check at least 1) condition is ready 2) generation=status.observedGeneration)